libcaf_core/caf/detail/parser/read_config.hpp

```diff
@@ -192,7 +192,7 @@ void read_config_map(State& ps, Consumer&& consumer) {
 template <class State, class Consumer>
 void read_config_uri(State& ps, Consumer&& consumer) {
   uri_builder builder;
-  auto g = make_scope_guard([&] {
+  auto g = make_scope_guard([&]() noexcept {
     if (ps.code <= pec::trailing_character)
       consumer.value(builder.make());
```

Comment:
Can throw, because it allocates memory. Maybe we can just add something to the testing DSL for this instead of using scope guards. Applies to a lot of other places as well, so let me think a bit about this.

Reply:
In the case we don't have enough memory here, we're having an problem anyways, whether we throw in the scope guard destructor or in an noexcept function. Not sure how to handle out-of-memory conditions, it falls back to the custom allocator and system resources.